src/utils/wallet.ts

```diff
@@ -333,6 +336,90 @@ const wallet = {
     const derivedXpub = xpub.deriveChild(derivationIndex);
     return derivedXpub.xpubkey;
   },
+
+  /**
+   * Create a P2SH MultiSig redeem script
+   *
+   * @param {string[]} xpubs The list of xpubkeys involved in this MultiSig
+   * @param {number} minSignatures Minimum number of signatures to send a
+   * transaction with this MultiSig
+   * @param {number} index Index to derive the xpubs
+   *
+   * @return {Buffer} A buffer with the redeemScript
+   * @throws {XPubError} In case any of the given xpubs are invalid
+   * @memberof Wallet
+   * @inner
+   */
+  createP2SHRedeemScript(xpubs, minSignatures, index) {
+    const sortedXpubs = _.sortBy(xpubs.map(xp => new HDPublicKey(xp)), (xpub) => {
+      return xpub.publicKey.toString('hex');
+    });
+
+    // xpub comes derived to m/45'/280'/0'
+    // Derive to m/45'/280'/0'/0/index
+    const pubkeys = sortedXpubs.map(xpub => xpub.deriveChild(0).deriveChild(index).publicKey);
+
+    // bitcore-lib sorts the public keys by default before building the script
+    // noSorting prevents that and keeps our order
+    const redeemScript = Script.buildMultisigOut(pubkeys, minSignatures, {noSorting: true});
+    return redeemScript.toBuffer();
+  },
+
+  /**
+   * Create a P2SH MultiSig input data from the signatures and redeemScript
+   *
+   * @param {Buffer[]} signatures The list of signatures collected from participants.
+   * @param {Buffer} redeemScript The redeemScript as a Buffer
+   *
+   * @return {Buffer} A buffer with the input data to send.
+   * @memberof Wallet
+   * @inner
+   */
+  getP2SHInputData(signatures, redeemScript) {
+    // minSignatures is the first opcode
+    // maxSignatures is the second to last opcode
+    const minSignatures = redeemScript.readUInt8() - OP_0.readUInt8();
+    const maxSignatures = redeemScript.readUInt8(redeemScript.length - 2) - OP_0.readUInt8();
+    if (signatures.length < minSignatures || signatures.length > maxSignatures) {
+      throw new Error('Signatures are incompatible with redeemScript');
+    }
+    const arr: Buffer[] = [];
+    for (const sig of signatures) {
+      helpers.pushDataToStack(arr, sig);
+    }
+    helpers.pushDataToStack(arr, redeemScript);
+    return util.buffer.concat(arr);
+  },
+
+  /**
+   * Create an HDPublicKey on P2SH MultiSig account path from the root xpub
```

Comment:
From the root xpriv*

Reply:
Fixed in [89d3a89](https://github.com/HathorNetwork/hathor-wallet-lib/pull/324/commits/89d3a89fb831ed94006445e3d29178170df31c29)